tests/data/generate.py

```diff
@@ -1,11 +1,21 @@
 #!/usr/bin/env python
 
+import sys
+
 import numpy as np
 import h5py
 
 
 def write_population_one_group(pop, prefix):
-    string_dtype = h5py.special_dtype(vlen=unicode)
+
+    if sys.version_info[0] == 2:
+        vlen_str_type = unicode
+    elif sys.version_info[0] == 3:
+        vlen_str_type = str
+    else:
+        raise 'Only python2/3 supported'
```

Comment:
You can avoid `sys` completely by just using:
```
try:
    vlen_str_type = unicode
except:
    vlen_str_type = str
```

Reply:
True, but this is more clear why we're doing it, and means that if unicode ever shows up again, it's future proof

Reply:
I would agree with @mgeplf here. And broad exceptions are also bug prone.

Reply:
Fair enough! The exception could be specialized… anyways, this could be simplified again once Py2 has died.

Reply:
yeah, that's the other reason I do it; easier to grep for `version_info` once we deprecate py2

Reply:
> The exception could be specialized…
Yes of course but you will have to use a NameError exception and I am always a bit reticent to use it in this particular use case (maybe I should not ...)